build/check.py

```diff
@@ -121,7 +121,8 @@ def get(*args):
 
   # Ignore missing goog.require since we assume the whole library is
   # already included.
-  opts = ['--jscomp_off=missingRequire', '--checks-only', '-O', 'SIMPLE']
+  opts = ['--jscomp_off=missingRequire', '--jscomp_off=lintChecks',
```

Comment:
Regrettably, the compiler contains two different variants of the "missing require" check (MISSING_REQUIRE_WARNING, MISSING_REQUIRE_CALL_WARNING). Some [legacy code](https://github.com/google/closure-compiler/blob/master/src/com/google/javascript/jscomp/DiagnosticGroups.java#L504) is preventing MISSING_REQUIRE_CALL_WARNING from being included with the "missingRequire" diagnostic group but rather it is included in lintChecks.

Said differently, all of the lintChecks besides MISSING_REQUIRE_CALL_WARNING still pass. Without adding lintChecks we get the following errors:

```
/home/ahochhaus/shaka-player2/demo/asset_section.js:30: ERROR - missing require: 'shakaAssets.testAssets'
  shakaAssets.testAssets.forEach(function(asset) {
  ^

/home/ahochhaus/shaka-player2/demo/asset_section.js:194: ERROR - missing require: 'shaka.log'
      shaka.log.debug('load() interrupted');
      ^

/home/ahochhaus/shaka-player2/demo/configuration_section.js:59: ERROR - missing require: 'shakaDemo.player_'
  shakaDemo.player_.configure(/** @type {shakaExtern.PlayerConfiguration} */({
  ^

/home/ahochhaus/shaka-player2/demo/configuration_section.js:71: ERROR - missing require: 'shakaDemo.controls_'
  shakaDemo.controls_.showTrickPlay(event.target.checked);
  ^

/home/ahochhaus/shaka-player2/demo/info_section.js:28: ERROR - missing require: 'shakaDemo.player_'
  shakaDemo.player_.addEventListener(
  ^

/home/ahochhaus/shaka-player2/demo/info_section.js:107: ERROR - missing require: 'shakaDemo.player_'
  var tracks = shakaDemo.player_.getTracks();
               ^

/home/ahochhaus/shaka-player2/demo/log_section.js:91: ERROR - missing require: 'shaka.log'
  shaka.log.setLevel(shaka.log.currentLevel);
  ^

/home/ahochhaus/shaka-player2/demo/main.js:79: ERROR - missing require: 'shaka.log'
    shaka.log.setLevel(shaka.log.Level.DEBUG);
    ^

/home/ahochhaus/shaka-player2/demo/main.js:84: ERROR - missing require: 'shaka.polyfill'
  shaka.polyfill.installAll();
  ^

/home/ahochhaus/shaka-player2/demo/main.js:86: ERROR - missing require: 'shaka.Player'
  if (!shaka.Player.isBrowserSupported()) {
       ^

/home/ahochhaus/shaka-player2/demo/main.js:112: ERROR - missing require: 'shaka.Player'
    shaka.Player.probeSupport().then(function(support) {
    ^

/home/ahochhaus/shaka-player2/test/cancelable_chain_unit.js:185: ERROR - missing require: 'shaka.test.Util'
      shaka.test.Util.delay(0.1).then(function() {
      ^

/home/ahochhaus/shaka-player2/test/dash_parser_content_protection_unit.js:33: ERROR - missing require: 'shaka.net.NetworkingEngine'
    var retry = shaka.net.NetworkingEngine.defaultRetryParameters();
                ^

/home/ahochhaus/shaka-player2/test/dash_parser_content_protection_unit.js:125: ERROR - missing require: 'shaka.util.Uint8ArrayUtils'
        initData: shaka.util.Uint8ArrayUtils.fromBase64(base64)
                  ^

/home/ahochhaus/shaka-player2/test/dash_parser_content_protection_unit.js:219: ERROR - missing require: 'shaka.test.Dash'
    var expected = shaka.test.Dash.makeManifestFromStreamSets([
                   ^

/home/ahochhaus/shaka-player2/test/dash_parser_live_unit.js:30: ERROR - missing require: 'shaka.net.NetworkingEngine'
    var retry = shaka.net.NetworkingEngine.defaultRetryParameters();
                ^

/home/ahochhaus/shaka-player2/test/dash_parser_live_unit.js:430: ERROR - missing require: 'shaka.util.StringUtils'
            var data = shaka.util.StringUtils.toUTF8(manifest);
                       ^

/home/ahochhaus/shaka-player2/test/dash_parser_live_unit.js:853: ERROR - missing require: 'shaka.test.Dash'
      shaka.test.Dash.makeReference('s4.mp4', 4, 30, 40)
      ^

/home/ahochhaus/shaka-player2/test/dash_parser_manifest_unit.js:27: ERROR - missing require: 'shaka.test.Dash'
    parser = shaka.test.Dash.makeDashParser();
             ^

/home/ahochhaus/shaka-player2/test/dash_parser_manifest_unit.js:555: ERROR - missing require: 'shaka.util.StringUtils'
          var data = shaka.util.StringUtils.toUTF8(source);
                     ^

/home/ahochhaus/shaka-player2/test/dash_parser_segment_base_unit.js:30: ERROR - missing require: 'shaka.test.Dash'
    parser = shaka.test.Dash.makeDashParser();
             ^

/home/ahochhaus/shaka-player2/test/dash_parser_segment_list_unit.js:30: ERROR - missing require: 'shaka.test.Dash'
    parser = shaka.test.Dash.makeDashParser();
             ^

/home/ahochhaus/shaka-player2/test/dash_parser_segment_list_unit.js:33: ERROR - missing require: 'shaka.test.Dash'
  shaka.test.Dash.makeTimelineTests('SegmentList', '', [
  ^

/home/ahochhaus/shaka-player2/test/dash_parser_segment_template_unit.js:30: ERROR - missing require: 'shaka.test.Dash'
    parser = shaka.test.Dash.makeDashParser();
             ^

/home/ahochhaus/shaka-player2/test/dash_parser_segment_template_unit.js:33: ERROR - missing require: 'shaka.test.Dash'
  shaka.test.Dash.makeTimelineTests(
  ^

/home/ahochhaus/shaka-player2/test/data_uri_plugin_unit.js:22: ERROR - missing require: 'shaka.net.NetworkingEngine'
    retryParameters = shaka.net.NetworkingEngine.defaultRetryParameters();
                      ^

/home/ahochhaus/shaka-player2/test/data_uri_plugin_unit.js:78: ERROR - missing require: 'shaka.util.StringUtils'
          var data = shaka.util.StringUtils.fromBytesAutoDetect(response.data);
                     ^

/home/ahochhaus/shaka-player2/test/data_uri_plugin_unit.js:87: ERROR - missing require: 'shaka.net.NetworkingEngine'
        shaka.net.NetworkingEngine.makeRequest([uri], retryParameters);
        ^

/home/ahochhaus/shaka-player2/test/data_uri_plugin_unit.js:88: ERROR - missing require: 'shaka.net.DataUriPlugin'
    shaka.net.DataUriPlugin(uri, request)
    ^

/home/ahochhaus/shaka-player2/test/db_engine_unit.js:31: ERROR - missing require: 'shaka.offline.DBEngine'
    shaka.offline.DBEngine.deleteDatabase().then(function() {
    ^

/home/ahochhaus/shaka-player2/test/db_engine_unit.js:117: ERROR - missing require: 'shaka.test.Util'
      shaka.test.Util.expectToEqualError(error, expectedError);
      ^

/home/ahochhaus/shaka-player2/test/db_engine_unit.js:126: ERROR - missing require: 'shaka.test.Util'
          return shaka.test.Util.delay(0.001);
                 ^

/home/ahochhaus/shaka-player2/test/drm_engine_integration.js:47: ERROR - missing require: 'shaka.media.DrmEngine'
    var supportTest = shaka.media.DrmEngine.probeSupport()
                      ^

/home/ahochhaus/shaka-player2/test/drm_engine_integration.js:73: ERROR - missing require: 'shaka.net.HttpPlugin'
      shaka.net.HttpPlugin(audioSegmentUri, dummyRequest)
      ^

/home/ahochhaus/shaka-player2/test/drm_engine_integration.js:282: ERROR - missing require: 'shaka.net.NetworkingEngine'
        retryParameters: shaka.net.NetworkingEngine.defaultRetryParameters(),
                         ^

/home/ahochhaus/shaka-player2/test/drm_engine_integration.js:350: ERROR - missing require: 'shaka.test.Util'
        shaka.test.Util.expectToEqualError(
        ^

/home/ahochhaus/shaka-player2/test/drm_engine_integration.js:386: ERROR - missing require: 'shaka.test.Util'
        return Promise.race([shaka.test.Util.delay(6), onErrorCalled]);
                             ^

/home/ahochhaus/shaka-player2/test/drm_engine_unit.js:99: ERROR - missing require: 'shaka.net.NetworkingEngine'
    var retryParameters = shaka.net.NetworkingEngine.defaultRetryParameters();
                          ^

/home/ahochhaus/shaka-player2/test/drm_engine_unit.js:563: ERROR - missing require: 'shaka.util.StringUtils'
        var initData = JSON.parse(shaka.util.StringUtils.fromUTF8(
                                  ^

/home/ahochhaus/shaka-player2/test/drm_engine_unit.js:898: ERROR - missing require: 'shaka.net.DataUriPlugin'
        return shaka.net.DataUriPlugin(request.uris[0], request);
               ^

/home/ahochhaus/shaka-player2/test/drm_engine_unit.js:910: ERROR - missing require: 'shaka.util.StringUtils'
            shaka.util.StringUtils.fromBytesAutoDetect(licenseBuffer);
            ^

/home/ahochhaus/shaka-player2/test/drm_engine_unit.js:942: ERROR - missing require: 'shaka.test.Util'
        shaka.test.Util.expectToEqualError(error, new shaka.util.Error(
        ^

/home/ahochhaus/shaka-player2/test/drm_engine_unit.js:1271: ERROR - missing require: 'shaka.test.Util'
        return shaka.test.Util.delay(0.1);
               ^

/home/ahochhaus/shaka-player2/test/ebml_parser_unit.js:238: ERROR - missing require: 'shaka.util.EbmlParser'
      shaka.util.EbmlParser.getVintValue_(data);
      ^

/home/ahochhaus/shaka-player2/test/fake_event_target_unit.js:132: ERROR - missing require: 'shaka.test.Util'
    shaka.test.Util.delay(0.1).then(function() {
    ^

/home/ahochhaus/shaka-player2/test/http_plugin_unit.js:48: ERROR - missing require: 'shaka.net.NetworkingEngine'
    retryParameters = shaka.net.NetworkingEngine.defaultRetryParameters();
                      ^

/home/ahochhaus/shaka-player2/test/http_plugin_unit.js:128: ERROR - missing require: 'shaka.net.NetworkingEngine'
    var request = shaka.net.NetworkingEngine.makeRequest(
                  ^

/home/ahochhaus/shaka-player2/test/http_plugin_unit.js:130: ERROR - missing require: 'shaka.net.HttpPlugin'
    shaka.net.HttpPlugin(uri, request)
    ^

/home/ahochhaus/shaka-player2/test/networking_engine_unit.js:49: ERROR - missing require: 'shaka.net.NetworkingEngine'
    shaka.net.NetworkingEngine.registerScheme('reject', rejectScheme);
    ^

/home/ahochhaus/shaka-player2/test/networking_engine_unit.js:239: ERROR - missing require: 'shaka.net.NetworkingEngine'
      shaka.net.NetworkingEngine.unregisterScheme('resolve');
      ^

/home/ahochhaus/shaka-player2/test/networking_engine_unit.js:630: ERROR - missing require: 'shaka.net.NetworkingEngine'
                          shaka.net.NetworkingEngine.defaultRetryParameters();
                          ^

/home/ahochhaus/shaka-player2/test/networking_engine_unit.js:631: ERROR - missing require: 'shaka.net.NetworkingEngine'
    return shaka.net.NetworkingEngine.makeRequest([uri], retryParameters);
           ^

/home/ahochhaus/shaka-player2/test/offline_integration.js:33: ERROR - missing require: 'shaka.Player'
    var supportPromise = shaka.Player.probeSupport()
                         ^

/home/ahochhaus/shaka-player2/test/offline_integration.js:41: ERROR - missing require: 'shaka.offline.DBEngine'
    Promise.all([shaka.offline.DBEngine.deleteDatabase(), supportPromise])
                 ^

/home/ahochhaus/shaka-player2/test/offline_integration.js:115: ERROR - missing require: 'shaka.test.Util'
      shaka.test.Util.expectToEqualError(e, expected);
      ^

/home/ahochhaus/shaka-player2/test/offline_integration.js:129: ERROR - missing require: 'shaka.test.Util'
          return shaka.test.Util.delay(5);
                 ^

/home/ahochhaus/shaka-player2/test/player_unit.js:120: ERROR - missing require: 'shaka.log'
        shaka.log.debug('finished unload 2');
        ^

/home/ahochhaus/shaka-player2/test/player_unit.js:253: ERROR - missing require: 'shaka.media.ManifestParser'
        shaka.media.ManifestParser.registerParserByMime('undefined', factory1);
        ^

/home/ahochhaus/shaka-player2/test/player_unit.js:335: ERROR - missing require: 'shaka.test.Util'
        shaka.test.Util.delay(0.5).then(function() {
        ^

/home/ahochhaus/shaka-player2/test/player_unit.js:1223: ERROR - missing require: 'shaka.test.Util'
        shaka.test.Util.expectToEqualError(
        ^

/home/ahochhaus/shaka-player2/test/simple_abr_manager_unit.js:278: ERROR - missing require: 'shaka.test.Util'
    var delay = shaka.test.Util.fakeEventLoop(
                ^

/home/ahochhaus/shaka-player2/test/storage_unit.js:636: ERROR - missing require: 'shaka.test.Util'
            shaka.test.Util.expectToEqualError(
            ^

/home/ahochhaus/shaka-player2/test/streaming_engine_integration.js:147: ERROR - missing require: 'shaka.test.StreamingEngineUtil'
      timeline = shaka.test.StreamingEngineUtil.createFakePresentationTimeline(
                 ^

/home/ahochhaus/shaka-player2/test/streaming_engine_integration.js:205: ERROR - missing require: 'shaka.test.StreamingEngineUtil'
        shaka.test.StreamingEngineUtil.boundsCheckPosition.bind(
        ^

/home/ahochhaus/shaka-player2/test/streaming_engine_integration.js:209: ERROR - missing require: 'shaka.test.StreamingEngineUtil'
        shaka.test.StreamingEngineUtil.getNumSegments.bind(
        ^

/home/ahochhaus/shaka-player2/test/streaming_engine_integration.js:214: ERROR - missing require: 'shaka.test.StreamingEngineUtil'
    netEngine = shaka.test.StreamingEngineUtil.createFakeNetworkingEngine(
                ^

/home/ahochhaus/shaka-player2/test/streaming_engine_integration.js:257: ERROR - missing require: 'shaka.test.StreamingEngineUtil'
    manifest = shaka.test.StreamingEngineUtil.createManifest(
               ^

/home/ahochhaus/shaka-player2/test/streaming_engine_integration.js:293: ERROR - missing require: 'shaka.net.NetworkingEngine'
      retryParameters: shaka.net.NetworkingEngine.defaultRetryParameters(),
                       ^

/home/ahochhaus/shaka-player2/test/streaming_engine_unit.js:232: ERROR - missing require: 'shaka.test.StreamingEngineUtil'
    timeline = shaka.test.StreamingEngineUtil.createFakePresentationTimeline(
               ^

/home/ahochhaus/shaka-player2/test/streaming_engine_unit.js:248: ERROR - missing require: 'shaka.test.StreamingEngineUtil'
    netEngine = shaka.test.StreamingEngineUtil.createFakeNetworkingEngine(
                ^

/home/ahochhaus/shaka-player2/test/streaming_engine_unit.js:266: ERROR - missing require: 'shaka.test.StreamingEngineUtil'
    manifest = shaka.test.StreamingEngineUtil.createManifest(
               ^

/home/ahochhaus/shaka-player2/test/streaming_engine_unit.js:304: ERROR - missing require: 'shaka.test.StreamingEngineUtil'
        shaka.test.StreamingEngineUtil.createMockVideoStream(6);
        ^

/home/ahochhaus/shaka-player2/test/streaming_engine_unit.js:1730: ERROR - missing require: 'shaka.net.NetworkingEngine'
        retryParameters: shaka.net.NetworkingEngine.defaultRetryParameters(),
                         ^

/home/ahochhaus/shaka-player2/test/support_check.js:20: ERROR - missing require: 'shaka.Player'
    if (!shaka.Player.isBrowserSupported()) {
         ^

/home/ahochhaus/shaka-player2/test/support_check.js:22: ERROR - missing require: 'shaka.test.Util'
      shaka.test.Util.cancelAllRemainingSpecs();
      ^

/home/ahochhaus/shaka-player2/test/util/dash_parser_util.js:26: ERROR - missing require: 'shaka.net.NetworkingEngine'
  var retry = shaka.net.NetworkingEngine.defaultRetryParameters();
              ^

/home/ahochhaus/shaka-player2/test/util/dash_parser_util.js:102: ERROR - missing require: 'shaka.util.StringUtils'
  var manifestData = shaka.util.StringUtils.toUTF8(manifestText);
                     ^

/home/ahochhaus/shaka-player2/test/util/dash_parser_util.js:110: ERROR - missing require: 'shaka.test.Util'
        shaka.test.Util.expectToEqualError(error, expectedError);
        ^

/home/ahochhaus/shaka-player2/test/util/manifest_generator.js:262: ERROR - missing require: 'shaka.util.Uint8ArrayUtils'
  var buffer = shaka.util.Uint8ArrayUtils.fromBase64(base64);
               ^

/home/ahochhaus/shaka-player2/test/util/manifest_generator.js:541: ERROR - missing require: 'goog.asserts'
  goog.asserts.assert(streamSet.streams.length > 0,
  ^

/home/ahochhaus/shaka-player2/test/util/memory_db_engine.js:85: ERROR - missing require: 'shaka.util.MapUtils'
  shaka.util.MapUtils.values(this.getStore_(storeName)).forEach(callback);
  ^

/home/ahochhaus/shaka-player2/test/util/memory_db_engine.js:135: ERROR - missing require: 'goog.asserts'
  goog.asserts.assert(storeName in this.stores_,
  ^

/home/ahochhaus/shaka-player2/test/util/stream_generator.js:482: ERROR - missing require: 'goog.asserts'
  goog.asserts.assert(
  ^

/home/ahochhaus/shaka-player2/test/util/stream_generator.js:500: ERROR - missing require: 'shaka.log'
    shaka.log.debug('tfdt box is version 1.');
    ^

/home/ahochhaus/shaka-player2/test/util/util.js:277: ERROR - missing require: 'shaka.log'
    shaka.log.setLevel(shaka.log.Level.INFO);
    ^

/home/ahochhaus/shaka-player2/test/util/util.js:279: ERROR - missing require: 'shaka.polyfill'
  shaka.polyfill.installAll();
  ^

/home/ahochhaus/shaka-player2/test/vtt_text_parser_unit.js:280: ERROR - missing require: 'shaka.util.StringUtils'
    var data = shaka.util.StringUtils.toUTF8(string);
               ^

/home/ahochhaus/shaka-player2/test/vtt_text_parser_unit.js:282: ERROR - missing require: 'shaka.media.VttTextParser'
      shaka.media.VttTextParser(data);
      ^

88 error(s), 0 warning(s)
```

Do you know of a better solution?